WordPress/Classes/Utility/BuildInformation/RemoteFeatureFlag.swift

```diff
@@ -0,0 +1,133 @@
+            return false
+        }
+    }
+
+    /// This key must match the server-side one for remote feature flagging
+    var remoteKey: String {
+        switch self {
+        case .jetpackFeaturesRemovalPhaseOne:
+            return "jp_removal_one"
+        case .jetpackFeaturesRemovalPhaseTwo:
+            return "jp_removal_two"
+        case .jetpackFeaturesRemovalPhaseThree:
+            return "jp_removal_three"
+        case .jetpackFeaturesRemovalPhaseFour:
+            return "jp_removal_four"
+        case .jetpackFeaturesRemovalPhaseNewUsers:
+            return "jp_removal_new_users"
+        case .jetpackFeaturesRemovalPhaseSelfHosted:
+            return "jp_removal_self_hosted"
+        case .jetpackFeaturesRemovalStaticPosters:
+            return "jp_removal_static_posters"
+        case .jetpackMigrationPreventDuplicateNotifications:
+            return "prevent_duplicate_notifs_remote_field"
+        case .wordPressSupportForum:
+            return "wordpress_support_forum_remote_field"
+        case .blaze:
+            return "blaze"
+        case .wordPressIndividualPluginSupport:
+            return "wp_individual_plugin_overlay"
+        }
+    }
+
+    var description: String {
+        switch self {
+        case .jetpackMigrationPreventDuplicateNotifications:
+            return "Jetpack Migration prevent duplicate WordPress app notifications when Jetpack is installed"
+        case .jetpackFeaturesRemovalPhaseOne:
+            return "Jetpack Features Removal Phase One"
+        case .jetpackFeaturesRemovalPhaseTwo:
+            return "Jetpack Features Removal Phase Two"
+        case .jetpackFeaturesRemovalPhaseThree:
+            return "Jetpack Features Removal Phase Three"
+        case .jetpackFeaturesRemovalPhaseFour:
+            return "Jetpack Features Removal Phase Four"
+        case .jetpackFeaturesRemovalPhaseNewUsers:
+            return "Jetpack Features Removal Phase For New Users"
+        case .jetpackFeaturesRemovalPhaseSelfHosted:
+            return "Jetpack Features Removal Phase For Self-Hosted Sites"
+        case .jetpackFeaturesRemovalStaticPosters:
+            return "Jetpack Features Removal Static Screens Phase"
+        case .wordPressSupportForum:
+            return "Provide support through a forum"
+        case .blaze:
+            return "Blaze"
+        case .wordPressIndividualPluginSupport:
+            return "Jetpack Individual Plugin Support for WordPress"
+        }
+    }
+
+    /// If the flag is overridden, the overridden value is returned.
+    /// If the flag exists in the local cache, the current value will be returned.
+    /// If the flag is not overridden and does not exist in the local cache, the compile-time default will be returned.
+    func enabled(using remoteStore: RemoteFeatureFlagStore = RemoteFeatureFlagStore(),
+                 overrideStore: FeatureFlagOverrideStore = FeatureFlagOverrideStore()) -> Bool {
```

Comment:
Injecting the stores makes the flags nice to used in the code (`if RemoteFeatureFlag.blaze.enabled() ...`) and leaves the API similar to the one for the build-time flags, which I think is good as we transition to remote flags.

However, I worry this will lead to unexpected behavior in the tests, where code is accessing one or more flags under the hood and it reads values that are not in line with the tests expectations (e.g. if something has been overridden).

How much friction do you think making both parameter required would add? It's a tough call to make. We want flags to be easy to use but also need to be mindful of testability because we want code to be thoroughly tested.

Reply:
Thanks for bringing this up! And I agree that this is indeed a tough call.

> I worry this will lead to unexpected behavior in the tests, where code is accessing one or more flags under the hood and it reads values that are not in line with the tests expectations (e.g. if something has been overridden).

`RemoteFeatureFlagStore` and `FeatureFlagOverrideStore` are built on user defaults. So in practice, any instance of them will return the same values unless it's custom a subclass.

> How much friction do you think making both parameter required would add? 

Personally, I see this as the responsibility of the test author. Dependency injection, in general, is a tool to enhance testability when needed, and since feature flags are a core feature, I think it's better to leave it optional, and folks should use it if their tests require it. 

Also, if we make the parameters required, folks can still choose not to inject their own stores in tests.